siac/hostcmd.go

```diff
@@ -51,6 +52,61 @@ Doing so will override the standard connectivity checks.`,
 	}
 )
 
+// parseSize converts strings of form 10GB to a size in bytes
+// If parsing fails the original value is returned unmodified
+func parseSize(value string) string {
+	var suffix string
+	var size int
+
+	// if the value is at least 2 or 3 characters it may have a suffix
+	// to detect a suffix a check is made that the character is outside
+	// the ascii range for numbers (48-57), letters are (65-90, 97-122)
+	if len(value) > 3 && value[len(value)-3] > 57 {
+		suffix = value[len(value)-3:] // KiB, MiB, GiB, TiB
+		_, err := fmt.Sscan(value[:len(value)-3], &size)
+		if err != nil {
+			return value
+		}
+	} else if len(value) > 2 && value[len(value)-2] > 57 {
+		suffix = value[len(value)-2:] // KB, MB, GB, TB
+		_, err := fmt.Sscan(value[:len(value)-2], &size)
+		if err != nil {
+			return value
+		}
+	} else {
+		// value does not appear to have a suffix, return as is
+		return value
+	}
+
+	switch strings.ToLower(suffix) {
+	case "tb":
```

Comment:
I chose ToLower so I wouldn't have to consider "Tb", "TIB", "Tib", etc. it seemed like there'd be too many variants to list explicitly and lowercase is cleaner